bsi/src/main/java/org/roaringbitmap/bsi/longlong/Roaring64NavigableMapSliceIndex.java

```diff
@@ -0,0 +1,622 @@
+package org.roaringbitmap.bsi.longlong;
+
+import org.roaringbitmap.bsi.BitmapSliceIndex;
+import org.roaringbitmap.bsi.Pair;
+import org.roaringbitmap.longlong.Roaring64NavigableMap;
+
+import java.io.*;
+import java.nio.ByteBuffer;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.IntStream;
+
+/*
+ * a Roaring64NavigableMap based bsi implementation
+ * The serialization format is compatible with bsi implementation of roaring(Go)
+ */
+public class Roaring64NavigableMapSliceIndex {
+  /**
+   * the maxValue of this bsi
+   */
+  private long maxValue;
+
+  /**
+   * the minValue of this bsi
+   */
+  private long minValue;
+
+  /**
+   * the bit component slice Array of this bsi
+   */
+  private Roaring64NavigableMap[] bA;
+
+  /**
+   * the exist bitmap of this bsi which means the columnId have value in this bsi
+   */
+  private Roaring64NavigableMap ebM;
+
+
+  private Boolean runOptimized = false;
+
+  /**
+   * NewBSI constructs a new BSI.  Min/Max values are optional.  If set to 0
+   * then the underlying BSI will be automatically sized.
+   */
+  public Roaring64NavigableMapSliceIndex(long minValue, long maxValue) {
+    if (minValue < 0) {
```

Comment:
Shoul we also check `maxValue >= minValue`?

Reply:
ok, should be fixed